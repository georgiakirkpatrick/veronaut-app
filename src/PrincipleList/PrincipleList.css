.PrincipleList {
    background-color: white;
    list-style-type: none;
    padding: 15vh 15vw 15vh 10vw;
    margin: 20vh 0;
    width: 100%;
    border: solid white;
    border-radius: 0 150px 150px 0px;
    height: 270vh;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
}

.PrincipleList h2 {
    padding: 3vh 0;
}

.PrincipleList header p {
    font: 400 16px/24px Poppins;
    letter-spacing: 1px;
    color: #041516;
}

.PrincipleList > ul {
    height: 80%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-end;
}

.PrincipleList > a {
    padding: 15px 0;
}

@media screen and (min-width: 360px) {
    .PrincipleList {
        height: 240vh;
        padding: 15vh 15vw 15vh 10vw;

    }
}

@media screen and (min-width: 375px) {
    .PrincipleList {
        height: 230vh;
        padding: 18vh 15vw 18vh 10vw;
    }
}

@media screen and (min-width: 375px) and (min-height: 812px) {
    .PrincipleList {
        height: 200vh;
        padding: 15vh 15vw 15vh 10vw;
    }
}

@media screen and (min-width: 411px) {
    .PrincipleList {
        margin: 20vh 0;
        height: 190vh;
        padding: 12vh 15vw 12vh 10vw;
    }
}

@media screen and (min-width: 540px) {
    .PrincipleList {
        margin: 20vh 0;
        height: 170vh;
        padding: 12vh 15vw 12vh 10vw;
    }
}

@media screen and (min-width: 768px) {
    .PrincipleList {
        height: 85vh;
        padding: 5vh;
        margin: 10vh 0;
    }

    .PrincipleList h2 {
        padding: 2vh 0;
    }
}

@media screen and (min-width: 1024px) {
    .PrincipleList {
        height: 120vh;
        padding: 10vh 8vw 10vh 5vw;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .PrincipleList {
        height: 70vh;
        padding: 7vh 12vw 7vh 8vw;
    }
}

@media screen and (min-width: 1440px) {
    .PrincipleList {
        padding: 10vh 10vw 10vh 5vw;
        margin: 20vh 0;
        height: 90vh;
    }    
}

@media screen and (min-width: 2560px) {
    .PrincipleList {
        padding: 5vh 10vw 5vh 5vw;
        margin: 10vh 0;
        height: 60vh;
    }
}