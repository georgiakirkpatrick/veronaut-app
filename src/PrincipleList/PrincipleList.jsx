import React from 'react'
import Principle from '../Principle/Principle'
import './PrincipleList.css'

const PrincipleList = props => {
  const {
    principles,
    readMore
  } = props

  const generatePrinciples = principles.map(principle => (
    <Principle
      key={principle.id}
      title={principle.title} 
      description={principle.description}
      symbol={principle.symbol}
    />
  ))
  
  return (
    <section className='PrincipleList'>
      <header>
        <h2>
          Transparency is key
        </h2>

        <p>
          Veronaut helps you understand product descriptions and points out the information brands choose to omit.
        </p>
      </header>

      <ul>
        {generatePrinciples}
      </ul>

      {readMore}
    </section>
  )
}

PrincipleList.defaultProps = {
  principles: [],
  readMore: ''
}

export default PrincipleList;