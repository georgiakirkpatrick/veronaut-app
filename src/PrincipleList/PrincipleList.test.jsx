import React from 'react';
import { render, screen } from '@testing-library/react';
import PrincipleList from './PrincipleList';
import { describe, it, expect } from 'vitest';

const principles = [
    {
        id: 1821,
        title: 'Good design', 
        description: 'We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.'
    },
    {
        id: 1822,
        title: 'Ethical',
        description: 'We report which brands have the most transparent supply chains, and which brands prioritize worker safety, living wages, and freedom of association (the right to join a union).'
    },
    {
        id: 1823,
        title: 'Sustainable',
        description: 'We feature clothing made with natural and cellulose-based fibers, with exceptions for a few categories like outerwear and swimwear.  While we include thoughtfully-sourced wool products, we exclude leather and fur.'
    }
]

describe ('PrincipleList', () => {
  it ('renders without crashing', () => {
    render(<PrincipleList />)
    screen.debug()
  })

  it('renders the UI as expected', () => {
    const tree = render(<PrincipleList principles={principles} />)
    expect(tree).toMatchSnapshot()    
  })
})