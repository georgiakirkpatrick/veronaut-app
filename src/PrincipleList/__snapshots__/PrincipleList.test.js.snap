// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrincipleList renders the UI as expected 1`] = `
<section
  className="PrincipleList"
>
  <header>
    <h2>
      Transparency is key
    </h2>
    <p>
      Veronaut helps you understand product descriptions and points out the information brands choose to omit.
    </p>
  </header>
  <ul>
    <li
      className="Principle"
    >
      <h3>
        Good design
      </h3>
      <p>
        We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.
      </p>
    </li>
    <li
      className="Principle"
    >
      <h3>
        Ethical
      </h3>
      <p>
        We report which brands have the most transparent supply chains, and which brands prioritize worker safety, living wages, and freedom of association (the right to join a union).
      </p>
    </li>
    <li
      className="Principle"
    >
      <h3>
        Sustainable
      </h3>
      <p>
        We feature clothing made with natural and cellulose-based fibers, with exceptions for a few categories like outerwear and swimwear.  While we include thoughtfully-sourced wool products, we exclude leather and fur.
      </p>
    </li>
  </ul>
  
</section>
`;
